src/smc-webapp/projects.cjsx

```diff
@@ -576,8 +576,9 @@ class ProjectsStore extends Store
         v = {}
         @get('project_map').map (project, project_id) =>
             upgrades = @getIn(['project_map', project_id, 'users', account_id, 'upgrades'])?.toJS()
-            if misc.len(upgrades)
-                v[project_id] = upgrades
+            for upgrade of upgrades
+                if upgrades[upgrade] > 0
+                    v[project_id] = upgrades
```

Comment:
Shouldn't you add a `break` line right directly below `v[project_id] = upgrades`.  Otherwise, it'll waste time doing `v[project_id] = upgrades` a whole bunch of times for no reason.

You could also do it like this (so use the val part of `of`):
```
for upgrade,val of upgrades
    if val > 0
        v[project_id] = upgrades
        break
```